History 페이지 추가

jihun1233 opened the pull request:
## 개요

- History page 추가
- FinishedTasks 컴포넌트 추가 - 완료된 Todo리스트 확인

## 작업 내용

- History 페이지의 레이아웃 UI 작업
- localStorage의 tasks를 가져와서 렌더링하는 기능 추가
- 삭제 기능은 완전하지 않음. 삭제 후 바로 렌더링 확인 불가

## 특이 사항

- 사항

## 스크린샷(or 동영상)

![image](https://user-images.githubusercontent.com/56039591/167064093-d9b2c8d1-97ce-4a5f-9ab4-2393cd786ef7.png)


## 기타

jihun1233 committed:
🐛 @typescript-eslint 관련 eslint 오류 수정

- 관련 rules를 찾을 수 없다는 에러메시지 확인

- 해당 모듈 설치하여 해결

jihun1233 committed:
✨History 컴포넌트 생성

- history 화면을 담당할 컴포넌트 생성

- 전체화면을 가정한 wrapper로 감싸놓음

jihun1233 committed:
Merge branch 'develop' of https://github.com/jihun1233/todo-list into feature-00-history

jihun1233 committed:
Merge branch 'develop' of https://github.com/jihun1233/todo-list into feature-00-history

jihun1233 committed:
➕ react icons 추가

jihun1233 committed:
🎨 Container route 바깥으로 이동

- 모든 페이지에 적용하기 때문에 route 밖에서 감싸줌

jihun1233 committed:
💄 History page에 사용되는 color, size 속성 추가

jihun1233 committed:
✨ HistoryPage 추가

- 기존의 History 폴더에서 HistoryPage로 이동 후 작업

- 주석은 li태그에 onClick을 사용했는데 이는 삭제기능 테스트용으로 삭제기능을 button으로 옮기면서 삭제할 것

jihyukma123 commented on package.json:
```diff
@@ -38,6 +39,9 @@
     ]
   },
   "devDependencies": {
+    "typescript": "^4.6.4",
+    "@typescript-eslint/eslint-plugin": "^5.22.0",
+    "@typescript-eslint/parser": "^5.22.0",
```
이 부분 혹시 eslintrc 파일에서 typescript 관련 설정 항목들 다 삭제해도 같은 오류가 뜨나요? 우선 저희 프로젝트에 타입스크립트를 사용하지 않는데 dependency  나 eslintrc 에 관련 항목이 있는게 맞는지 잘 모르겠다는 생각이 들기도 합니다.. 번거로우시겠지만 혹시 typescript 관련 설치 파일 및 eslintrc 에서 관련 항목들 삭제한 후에도 같은 오류가 뜨는지 한 번 확인해주실 수 있을까요..?

jihun1233 committed:
🔧 typescript 관련 eslint 설정 제거

- 기존에 모듈을 설치해서 해결했던 부분을 typescript를 사용하지 않아 eslint에서 관련 설정을 모두 제거하는것으로 변경

jihyukma123 commented on src/routes/historyPage/index.js:
```diff
@@ -1,5 +1,61 @@
+import { useEffect, useState } from 'react'
+import style from './HistoryPage.module.scss'
```
사소한 거긴 한데 코드를 통일하는 것이 좋을 듯 해서.... 다른 분들 pr 보니까 다들 
```javascript
import styles from './HistoryPage.module.scss'
```
이런 식으로 `styles` 로 쓰셔서 지훈님께서도 이 부분 수정해주시면 좋을 듯 합니다!(뭔가 더 큰 뜻이 있으신거라면 죄송합니다..)

자꾸 지훈님께 수정사항을 말씀드리네요 공교롭게도..번거롭게 해드려서 죄송합니다..!

jihun1233 replied:
아니에요 괜찮아요!
컴포넌트 분할 작업 중이라 styles 변경 포함해서 같이 커밋할게요

jihun1233 committed:
🎨 FinishedTasks 컴포넌트 분리

- HistoryPage에 모두 작성되었던 내용을 FinishedTasks컴포넌트를 생성하여 분리함

- HistoryPage에는 페이지 전체를 나타내는 레이아웃이 남도록 하고, 여기에 페이지를 구성하는 컴포넌트를 추가
- scss 모듈을 가져오는 이름을 style -> styles로 변경

markyul commented on src/routes/historyPage/FinishedTasks.js:
```diff
@@ -0,0 +1,53 @@
+  )
+}
+
+export default FinishedTasks
```
이 파일을 그냥 index 안에 HistoryPage 컴포넌트 밑에 컴포넌트로 하나 만들어서 사용하는건 어떨까요?
그게 통일성이 있을 것 같습니다.

jihun1233 replied:
페이지를 깔끔하게 유지하고 컴포넌트를 나누는게 좋겠다고 생각했습니다! 그리고 통계 잔디도 들어올거라 한번에 담으면 너무 커질것같아서요

markyul replied:
알겠습니다!

jihun1233 committed:
➕ classnames 추가

jihun1233 committed:
✨ category 색 circle icon 추가

- 카테고리별 색상에 맞게 circle모양 아이콘 추가

- 카테고리별 classname을 가져올 수 있도록 findCategoryClassname 추가

jihun1233 committed:
Merge branch 'develop' of https://github.com/POB-Frontend-4th-6team/todo-list into feature-00-history

jihun1233 committed:
🐛 중복 선언 제거

- merge 후 Container가 중복 선언되어 해결

jihun1233 committed:
Merge branch 'develop' into feature-00-history
